### PR TITLE
ticket 5 finished

### LINKED ISSUE #5: Can't remove an item from the cart
Customers are complaining that they can't remove an item from the cart after it has been added.

When the client performs a DELETE request to the `/lineitems/n` resource, it should remove the specified item from the user's cart.

### PR DESCRIPTION
**Pull request by Kamiran79**
Description of PR that completes issue here...

## Changes

- Item 1: line 70 comment don't needed on the file views\lineitem.py
- Item 2: add two lines 80, 81 on the file views\lineitem.py
- Item 3: the add was instead the order_product change it to lineItems for line 80
- item 4: added lineItems.delete() on line 81 to do the delete.

## Requests / Responses

If this PR contains code that defines a new request/response, or changes an existing one, please put the JSON representations here.

**Request**

```
GET `profile/cart` Checking what line items there

```json
{
    "id": 13,
    "url": "http://localhost:8000/orders/13",
    "created_date": "2021-02-12",
    "payment_type": null,
    "customer": "http://localhost:8000/customers/7",
    "lineitems": [
        {
            "id": 14,
            "product": {
                "id": 88,
                "name": "Element",
                "price": 1727.41,
                "number_sold": 0,
                "description": "2003 Honda",
                "quantity": 3,
                "created_date": "2019-05-28",
                "location": "Dukoh",
                "image_path": null,
                "average_rating": -1
            }
        },
        {
            "id": 16,
            "product": {
                "id": 88,
                "name": "Element",
                "price": 1727.41,
                "number_sold": 0,
                "description": "2003 Honda",
                "quantity": 3,
                "created_date": "2019-05-28",
                "location": "Dukoh",
                "image_path": null,
                "average_rating": -1
            }
        }
    ],
    "size": 2
}
```
GET `/lineitems/16` Checking that item is there

```json
{
    "id": 16,
    "url": "http://localhost:8000/lineitems/16",
    "order": "http://localhost:8000/orders/13",
    "product": "http://localhost:8000/products/88"
}
```
DELETE  `/lineitems/16` want to delete that items

```json
{
}
```
GET `profile/cart` Checking what is in the cart .. noticed that still there. after update the code tested it worked.

```json
{
    "id": 13,
    "url": "http://localhost:8000/orders/13",
    "created_date": "2021-02-12",
    "payment_type": null,
    "customer": "http://localhost:8000/customers/7",
    "lineitems": [
        {
            "id": 14,
            "product": {
                "id": 88,
                "name": "Element",
                "price": 1727.41,
                "number_sold": 0,
                "description": "2003 Honda",
                "quantity": 3,
                "created_date": "2019-05-28",
                "location": "Dukoh",
                "image_path": null,
                "average_rating": -1
            }
        },
        {
            "id": 16,
            "product": {
                "id": 88,
                "name": "Element",
                "price": 1727.41,
                "number_sold": 0,
                "description": "2003 Honda",
                "quantity": 3,
                "created_date": "2019-05-28",
                "location": "Dukoh",
                "image_path": null,
                "average_rating": -1
            }
        }
    ],
    "size": 2
}
```

**Response**

"GET /profile/cart HTTP/1.1" 200 OK
"DELETE /lineitems/16 HTTP/1.1" 204 OK
Order object (13) OK
"GET /profile/cart HTTP/1.1" 200 OK
Not Found: /lineitems/16
"GET /lineitems/16 HTTP/1.1" 404

```json
{
    "id": 54,
    "title": "Kite",
    "product_type_id": 1,
    "description": "Red. It flies high.",
    "quantity": 5
}
```

## Testing

Description of how to test code...

- [ ] Run GET http://localhost:8000/profile/cart - to see what is in the cart
- [ ] Run add new item to cart just to make sure have more lineitems POST http://localhost:8000/profile/cart
- [ ] Run GET http://localhost:8000/lineitems/16
- [ ] Run DELETE http://localhost:8000/lineitems/16
- [ ] Run GET http://localhost:8000/lineitems/16 noticed the item is still there
- [ ] Run GET http://localhost:8000/profile/cart - to see what is in the cart - just to make sure that still there
- [ ] Run noticed the destroy code on views\lineitems.py was no delete for lineitems, I added that two lines on 80 and 81
- [ ] Did comment line 79 just for reference
- [ ] Run the DELETE http://localhost:8000/lineitems/16
- [ ] Run GET http://localhost:8000/lineitems/16 got me error  "message": "OrderProduct matching query does not exist."
- [ ] Run GET http://localhost:8000/profile/cart - to see what is in the cart, noticed that 16 got deleted.

## Related Issues

- Fixes #5

### CODE REVIEW

File: bangazonapi/views/lineitem.py [hunk 1]
```diff
@@ -76,7 +76,9 @@ def destroy(self, request, pk=None):
         """
         try:
             customer = Customer.objects.get(user=request.auth.user)
-            order_product = OrderProduct.objects.get(pk=pk, order__customer=customer)
+            #order_product = OrderProduct.objects.get(pk=pk, order__customer=customer)
```
Comment (hannahhall) [inline bangazonapi/views/lineitem.py]: Since this line doesn't need to be here it should be deleted instead of commented out

**Review by hannahhall — CHANGES_REQUESTED** [review]

**Comment by hannahhall** [comment]
Also the testing steps in the pr description are for a different ticket than what the code is fixing

**Comment by Kamiran79** [comment]
I update that delete the comment line on line 79, Also did update the title to match the correct ticket was 4 made it to 5.

**Review by hannahhall — APPROVED** [review]
👍

**Comment by Kamiran79** [comment]
fixed #5

**Pull request merged**